### ISSUE TITLE
web-ui query: oversized response

### ISSUE THREAD

**Issue by vron** [perkeep/perkeep#1209]
Originally raised at: https://groups.google.com/forum/#!topic/camlistore/x94Udt2cNyg

after:

pk-put file --permanode --title "Backup" $HOME

and visiting the web-ui I get a response from the /query endpoint of 20Mb+ in size.

The issue seems to be that the entire recursive structure under the new permanode is returned when only the permanode needs to be shown.

**Comment by mpl** [comment]
@vron the attached screenshot is what the query at the top level of the web UI looks like, when one of the elements in the permanode for a dir of ~60MB. As you can see the response to the query is apparently ~35KB for me in this case, which seems reasonable.

Can you tell me how you instrumented to arrive at 20MB please? What does it look like when you use the chrome dev tools (network tab), like I did?

![screenshot](https://user-images.githubusercontent.com/626071/44360961-b4dd2980-a4bc-11e8-9e01-2721d535328b.png)

**Comment by vron** [comment]
Hi, i see it when displaying in the "list" view, see for example:

![screen shot 2018-08-20 at 09 56 31](https://user-images.githubusercontent.com/160233/44361042-fe2d7900-a4bc-11e8-9c89-e9eea0e9721c.png)

**Comment by vron** [comment]
Chrome wont let me look a tthe response - (to large?) but when cURL:ing it I see a json file of 20Mb with the entire directory tree.

Please note that such a Backup permanode is a recursive directory with lots of files and representing a total directory size in the order of 100s of GBs

**Comment by mpl** [comment]
Ok, I understand what's going on.
The describe rules are usually what controls the (recursive) expansion of the description of any blob found by the search query.

However, as you see at https://perkeep.org/pkg/search#DescribeRule , all the mechanisms in the rules only apply to permanodes and their attributes. Which means, in the case of a top permanode for a directory, the rules do not offer any way of describing further than the first description.

So with any search query, there would be no way to "drill down" into directories. And it happens that sometimes, you do want that; for example when you want to zip everything under a directory for a download. Which is why I probably at some point elected to rely on the other parameter that influences descriptions ("depth"), in order to let things infinitely recurse when describing directories.

Anyway, TL;DR is you can probably apply https://perkeep-review.googlesource.com/c/perkeep/+/18226 to fix your immediate problem, but I don't think it's the complete solution, so I have to think more about it.

**Comment by vron** [comment]
Thx

**Comment by mpl** [comment]
I think I'm now approaching a much better fix, which removes - long overdue - the Depth parameter altogether. CL is updated.
